### ISSUE TITLE
@Embedded element not retrieved (null) when it contains a list

### ISSUE THREAD

**Issue by bejayoharen** [Impetus/kundera#297]
I have:

```
@Entity(name = "XonamiUser")
@Table(name = "XonamiUser", schema = "xonami@mongo")
public class XonamiUser implements PropertyContainer.SearchTermProvider, GenericDataProducer, Serializable {
@Embedded
private PropertyContainer propertyContainer;
...
```

and:

```
@Embeddable
public final class PropertyContainer  {
public interface SearchTermProvider {
    public ArrayList<String> getAdditionalSearchTerms();
}
@Column
private String hi;
@Column
private List<String> tags;
@Column
private List<String> propertyKeys ;
@Column
private List<String> propertyValues ;
@Column
protected List<String> searchList;

@Transient
private SearchTermProvider searchTermProvider;

public PropertyContainer() {
    tags            = new LinkedList<String>();
    propertyKeys    = new LinkedList<String>();
    propertyValues  = new LinkedList<String>();
    searchList      = new LinkedList<String>();
    this.searchTermProvider = null;
    tags.add( "yo" );
    propertyKeys.add( "yo" );
    propertyValues.add( "hey" );
    tags.add( "yo" );
    hi = "hello";
}
...
```

When I retrieve XonamiUser from mongo, PropertyContainer is null. Moreover, propertyKeys, propertyValues, and Tags are not written to the database. If I comment out references to Lists, it is not null when I retrieve from the database.

**Comment by mevivs** [comment]
This should be working. Let me verify and update you on this.

-Vivek

**Comment by mevivs** [comment]
Confirming this is an issue, working on it. 

However as a work around, if you move :

```
@Column
private List<String> tags;
@Column
private List<String> propertyKeys ;
@Column
private List<String> propertyValues ;
@Column
protected List<String> searchList;
```

with XonamiUser entity, it will work.

-Vivek

**Comment by bejayoharen** [comment]
That's what I'm doing, but it's not how I want it. (I actually want XonamiUser to extend PropertyContainer, but that's not supported.)

I'd also like tags and searchList to be sets, but I can't get that to work either. Do you have sample code for that?

**Comment by mevivs** [comment]
Let me fix it for Embeddable itself and share junits with you.

-Vivek

**Comment by bejayoharen** [comment]
unfortunately, when I try an operation like

```
tags.clear();
```

I get 

```
java.lang.UnsupportedOperationException: null
 at java.util.AbstractList.remove (AbstractList.java: 144)
 at java.util.AbstractList$Itr.remove (AbstractList.java: 360)
 at java.util.AbstractList.removeRange (AbstractList.java: 559)
 at java.util.AbstractList.clear (AbstractList.java: 217)
 at com.xonami.rest.db.PropertyHelper.updateSearchList (P
```

Are lists built by Kundera from the database not editable?

Possibly related, I also see warnings like this, albeit in prior calls:

```
Jun 12, 2013 3:26:05 PM com.impetus.kundera.property.accessor.ObjectAccessor getCopy
WARNING: Object of class class java.util.LinkedList can't be cloned, due to exception:java.lang.ClassCastException@43f854bd
```

**Comment by mevivs** [comment]
It shouldn't be. i am looking into this.

-Vivek

**Comment by kkmishra** [comment]
Fix has been added for this, please verify.

Thanks
Kuldeep

**Comment by mevivs** [comment]
Fixed and releasing with 2.6. Please verify.
-Vivek

**Comment by bejayoharen** [comment]
I can't verify, as I'm no longer using kundera.
